### PR TITLE
Simplify FontProperties init.

### PR DESCRIPTION
**Pull request by anntzer**
The various setters already set the None attributes to the rcParams
values; no need to do this twice in `__init__`.

Also inline the now-single-use _normalize_font_family, move the aliases
all to a single block, and prefer get_size to get_size_in_points.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

**Comment by timhoffm** [comment]
Since you want to get rid of the setters and make font properties immutable, isn't this going in the wrong direction?

**Comment by anntzer** [comment]
I thought that switching to a fully immutable FP would anyways require moving all the validation logic into `__init__` (it isn't there for now), which is a bigger change not made easier or harder by the current PR.  The point was more to get rid of _normalize_font_family, in fact...

File: lib/matplotlib/font_manager.py [hunk 1]
```diff
@@ -967,6 +954,12 @@ def copy(self):
         """Return a copy of self."""
         return copy.copy(self)
 
+    # Aliases.
```
Comment (timhoffm) [inline lib/matplotlib/font_manager.py:957]:
```suggestion
    # Aliases
```

**Review by timhoffm — APPROVED** [review]

File: lib/matplotlib/font_manager.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/font_manager.py]: sure.

**Review by oscargus — APPROVED** [review]

**Pull request merged**